### DISCUSSION TITLE
Call a route as a data route from the server?

### DISCUSSION THREAD

**Discussion by acoreyj** [remix-run/remix#5258] [Q&A]
I recently added some UI to a resource route but now the fetch I am doing that route returns html instead of what is in the action. Looking through the code I see remix-react and react-router use `_data` params with the route id to call routes without getting html but it doesn't seem like that is officially documented and I don't see any way of getting the route id from the server side.

Is this possible? If not is it a feature you would be interested in having, I could implement with thoughts on best way to do it. Perhaps sending the routeId to loaders/actions and documenting the _data search param.

**Comment by machour** [comment]
Hey, if your resource route starts outputting HTML, it may be because you have a default export in it. 

_data is meant to be used internally by Remix, and you should never need to use it directly
